### PR TITLE
shapeless-based derivation

### PR DESCRIPTION
**Pull request by senia-psm**
Create `zio-config-shapeless` as a drop-in replacement for `zio-config-magnolia`.
Unlike `magnolia` it supports all required `scala` versions.

### CODE REVIEW

**Comment by afsalthaj** [comment]
@senia-psm thanks for this. Great addition. we will keep both zio-config-magnolia and zio-config-shapeless.

**Comment by senia-psm** [comment]
@afsalthaj could you please update docs? You can specify that `zio-config-magnolia` is the reference (default) implementation of derivation.

**Comment by afsalthaj** [comment]
Yes  I will !

File: shapeless/src/test/scala/zio/config/shapeless/AutomaticConfigDescriptorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+  val genLocalDateString: Gen[Random with Sized, String] =
+    genInstant.map(_.atZone(ZoneOffset.UTC).toLocalDate.toString)
+
+  val genLocalDateTimeString: Gen[Random with Sized, String] =
+    genInstant.map(_.atZone(ZoneOffset.UTC).toLocalDateTime.toString)
+
+  val genLocalTimeString: Gen[Random with Sized, String] =
+    genInstant.map(_.atZone(ZoneOffset.UTC).toLocalTime.toString)
+}
```
Comment (afsalthaj) [inline shapeless/src/test/scala/zio/config/shapeless/AutomaticConfigDescriptorSpec.scala:154]: 👍

Comment (senia-psm) [reply shapeless/src/test/scala/zio/config/shapeless/AutomaticConfigDescriptorSpec.scala:154]: All the tests are just copied from `magnolia`

**Pull request merged**